File: Code/Asteroid_Trajectory_Selection/asteroid_selector.m
clear;

try
    asteroid_list = LOAD_KERNELS("NOTABLE_ASTEROID_BSPs");
catch exception
    close all force;
    fprintf('An error occurred: %s\n', exception.message);
    error("Most likely you need to add NOTABLE_ASTEROID_BSPs to your path" + ...
            " and you must be in the main github folder.");
end

MAX_M = 2;
MIN_M = 0;

MIN_LAUNCH_DATE_UTC = 'Jan 1 12:00:00 UTC 2027';
MAX_LAUNCH_DATE_UTC = 'Dec 31 12:00:00 UTC 2030';

for m_1 = MIN_M:MAX_M
    for m_2 = MIN_M:MAX_M
        for m_3 = MIN_M:MAX_M
            GENERATE_OPTIMIZED_DATA(asteroid_list, m_1, m_2, m_3, MIN_LAUNCH_DATE_UTC, MAX_LAUNCH_DATE_UTC);
        end
    end
end

File: Python/asteroid_trajectory_selection/asteroid_selector.py
"""Asteroid selector script.

Loads SPICE kernels, sets launch-date parameters, and calls
generate_optimized_data to compute optimal trajectories.

Translates asteroid_selector.m to Python.
"""

import sys

from .load_kernels import load_kernels
from .generate_optimized_data import generate_optimized_data


def main():
    bsp_folder = "NOTABLE_ASTEROID_BSPs"
    generic_kernels = "generic_kernels"

    try:
        asteroid_list = load_kernels(bsp_folder, generic_kernels)
    except Exception as e:
        print(f"An error occurred: {e}")
        print("Most likely you need to add NOTABLE_ASTEROID_BSPs to your path "
              "and you must be in the main github folder.")
        sys.exit(1)

    MAX_M = 2
    MIN_M = 0

    MIN_LAUNCH_DATE_UTC = 'Jan 1 12:00:00 UTC 2027'
    MAX_LAUNCH_DATE_UTC = 'Dec 31 12:00:00 UTC 2035'

    asteroid_optimized_data = generate_optimized_data(
        asteroid_list, 0, 0, 0,
        MIN_LAUNCH_DATE_UTC, MAX_LAUNCH_DATE_UTC
    )

    return asteroid_optimized_data


if __name__ == '__main__':
    main()
